### DISCUSSION TITLE
AvaloniaScheduler should not be defined in Avalonia.ReactiveUI

### DISCUSSION THREAD

**Discussion by SuperJMN** [AvaloniaUI/Avalonia#15034] [General]
The [AvaloniaScheduler](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.ReactiveUI/AvaloniaScheduler.cs) is a **Scheduler**, not a ReactiveUI feature. However, it's included in the ReactiveUI package. This is wrong in my opinion.

Let my use case serve as a sample. I'm creating a Templated Control for my library and I'm using System.Reactive. I need to specify the correct scheduler and I've been forced to use Avalonia.ReactiveUI for this. But my library doesn't use ReactiveUI at all.

It should be moved to a more generic package. Maybe Avalonia core package.

**Comment by stevemonaco** [comment]
As a workaround, you can copy/paste the `AvaloniaScheduler` source into your project (changing class name and/or namespace), add `System.Reactive` package, and it should work without pulling in a ReactiveUI dependency. There aren't any internals referenced.

**Comment by maxkatz6** [comment]
> It should be moved to a more generic package. Maybe Avalonia core package.

It would require a System.Reactive dependency just for that.
At the same time, you can reference Avalonia.ReactiveUI package, but not use reactive ui types. 
That's a tradeoff we have now.

**Comment by stevemonaco** [comment]
`AvaloniaScheduler` and `AvaloniaObjectReactiveExtensions` could be moved into a new `Avalonia.Reactive` package (not a core package), but it is another package to maintain.
